### PR TITLE
cert rotation

### PR DESCRIPTION
**Pull request by marcolan018**
https://github.com/open-cluster-management/backlog/issues/11157

support certificate rotation, not include CA rotation, which will be included in another pr

Signed-off-by: Marco <llan@redhat.com>

### CODE REVIEW

File: pkg/certificates/cert_controller.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	}
+	watchlist := cache.NewListWatchFromClient(kubeClient.CoreV1().RESTClient(), "secrets", config.GetDefaultNamespace(),
+		fields.Everything())
+	_, controller := cache.NewInformer(
+		watchlist,
+		&v1.Secret{},
+		time.Minute*60,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				restartPods(*kubeClient, *obj.(*v1.Secret))
```
Comment (clyang82) [inline pkg/certificates/cert_controller.go:56]: the rbac-query-proxy and  observatorium-api  will be restarted for the first time when the secrets are created. is it  desired behaviour?

File: pkg/certificates/cert_controller.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+			},
+
+			DeleteFunc: func(obj interface{}) {
+			},
```
Comment (clyang82) [inline pkg/certificates/cert_controller.go:60]: I think we should restart the pod if the secrets are deleted, because the secrets will be recreated. WDYT?

Comment (marcolan018) [reply pkg/certificates/cert_controller.go:60]: the restart will happen in Add event when the certificates recreated after deletion

File: pkg/certificates/cert_controller.go (hunk 1 above)
Comment (marcolan018) [reply pkg/certificates/cert_controller.go:56]: in function updateDeployLabel, the restart label will only be injected when secret is created after deployment. So for first time,  there will be no restart.

Comment (clyang82) [reply pkg/certificates/cert_controller.go:56]: 👍 You can follow the same way of cert-manager does.

**Pull request merged**